ports/esp32/esp32_common.cmake

```diff
@@ -52,6 +52,45 @@ list(APPEND MICROPY_SOURCE_DRIVERS
     ${MICROPY_DIR}/drivers/dht/dht.c
 )
 
+if(MICROPY_PY_TINYUSB)
+    string(CONCAT GIT_SUBMODULES "${GIT_SUBMODULES} " lib/tinyusb)
+
+    if(ECHO_SUBMODULES)
+    # No-op, we're just doing submodule/variant discovery.
+    # Cannot run the add_library/target_include_directories rules (even though
+    # the build won't run) because IDF will attempt verify the files exist.
```

Comment:
There's an obscure failure mode here where if someone does:

```
make submodules
make BOARD=ESP32_GENERIC_S2
```

then the build may fail, because the submodules step didn't fetch TinyUSB. (I note the ci.sh script needed to be updated not to trigger exactly this failure mode!)

Suggest we always append lib/tinyusb to the submodules list for esp32 port. The downside is small, this check and explanation can be removed, and I don't think anyone will ever notice they got a maybe-unnecessary submodule (but definitely someone somewhere will avoid running into a weird build error).

Reply:
This is a failure case on many ports for a number of submodules, `nimble` is a common one for that too. But yeah I'm happy to always include it just to simplify things!